### PR TITLE
add compatibility tests for v7.1.x

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Adds compatibility tests for v7.1.x.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
N/A
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/compatibility-test-matrices/release-v7.1.x/client-chain-b.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "chain-a": ["release-v7.1.x", "v6.1.0", "v5.2.0", "v4.3.0", "v4.2.0", "v4.1.1", "v3.4.0", "v3.3.1"],
+  "chain-b": ["release-v7.1.x"],
+  "entrypoint": ["TestClientTestSuite"],
+  "test": [
+    "TestClientUpdateProposal_Succeeds"
```
Comment (chatton) [inline .github/compatibility-test-matrices/release-v7.1.x/client-chain-b.json:6]: we will need to add the misbehaviour test when it is merged I will try to remember this before merging!

File: .github/compatibility-test-matrices/release-v7.1.x/ica-chain-a.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "chain-a": ["release-v7.1.x"],
+  "chain-b": ["release-v7.1.x", "v6.1.0"],
```
Comment (chatton) [inline .github/compatibility-test-matrices/release-v7.1.x/ica-chain-a.json:3]: should we add a v7.0.0 rc? to these lists?

**Review by chatton — APPROVED** [review]
LGTM!  Thanks for keeping these up to date. I just had the one question about adding v7 to some of these.

**Review by colin-axner — APPROVED** [review]
LGTM. Agree with @chatton's comments

**Review by damiannolan — APPROVED** [review]

File: .github/compatibility-test-matrices/release-v7.1.x/client-chain-b.json (hunk 1 above)
Comment (chatton) [reply .github/compatibility-test-matrices/release-v7.1.x/client-chain-b.json:6]: feel free to merge whenever you're ready and I'll add this entry to my PR!

**Pull request merged**